service/s3/s3manager/download_test.go

```diff
@@ -644,6 +646,80 @@ func TestDownload_WithFailure(t *testing.T) {
 	}
 }
 
+func TestDownloadBufferStrategy(t *testing.T) {
```

Comment:
It would be helpful to pull in the s3manager v1 test updates made in https://github.com/aws/aws-sdk-go/pull/2841 to this PR as well. The `fillRandom` function took enough time it could cause Go tests to sometimes timeout with 10m when running in Travis.

Reply:
Added